### PR TITLE
Add gsl-lite

### PR DESCRIPTION
**Pull request by tadeu**
@conda-forge/help-c-cpp, ready for review :)

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
I rerun in azure the ``osx`` just to investigate why it is failing there

File: recipes/gsl-lite/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - {{ compiler('cxx') }}
+    - make  # [unix]
+  files:
+    - test
+
+  commands:
+    - cd test
+    - cmake . -DCMAKE_BUILD_TYPE=Release
+    - cmake --build . --config Release
+    - ./program  # [unix]
```
Comment (marcelotrevisani) [inline recipes/gsl-lite/meta.yaml]: Please, try to add permission to execute this, because it may be the problem in azure
```
chmod +x program  
```

**Comment by marcelotrevisani** [comment]
This is the error
```
+ cmake --build . --config Release
ld: warning: ignoring file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd, file was built for unsupported file format ( 0x2D 0x2D 0x2D 0x20 0x21 0x74 0x61 0x70 0x69 0x2D 0x74 0x62 0x64 0x2D 0x76 0x33 ) which is not the architecture being linked (x86_64): /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd
ld: symbol dyld_stub_binder not found (normally in libSystem.dylib).  Needed to perform lazy binding to function __ZSt9terminatev for architecture x86_64
clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/program.dir/build.make:84: program] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/program.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```
I think you may specify the architecture

**Comment by marcelotrevisani** [comment]
cc: @ocefpaf

**Comment by tadeu** [comment]
It's passing in Travis (standard macOS)... since we were already ignoring azure-pipelines in win64 (because of problems there), can we also ignore azure-pipelines in macOS?

File: recipes/gsl-lite/meta.yaml (hunk 1 above)
Comment (tadeu) [reply recipes/gsl-lite/meta.yaml]: This wasn't needed in Linux (CMake+make+gcc already generates an executable with executable permission)... is it really needed for macOS?

Comment (marcelotrevisani) [reply recipes/gsl-lite/meta.yaml]: It is not necessary I was just doing some tests

**Comment by marcelotrevisani** [comment]
@tadeu probably there is something missing in the azure pipeline,
I will revert my modification and merge it.

**Comment by tadeu** [comment]
@marcelotrevisani, yeah, thanks :D

**Pull request merged**